warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -218,232 +222,394 @@ func (*AzureSynapse) IsEmpty(context.Context, model.Warehouse) (empty bool, err
-					} else {
-						finalColumnValues = append(finalColumnValues, convertedValue)
-					}
-				case "datetime":
-					var convertedValue time.Time
-					// TODO : handling milli?
-					if convertedValue, err = time.Parse(time.RFC3339, strValue); err != nil {
-						as.logger.Errorf("AZ : Mismatch in datatype for type : %s, column : %s, value : %s, err : %v", valueType, sortedColumnKeys[index], strValue, err)
-						finalColumnValues = append(finalColumnValues, nil)
-					} else {
-						finalColumnValues = append(finalColumnValues, convertedValue)
-					}
-					// TODO : handling all cases?
-				case "boolean":
-					var convertedValue bool
-					if convertedValue, err = strconv.ParseBool(strValue); err != nil {
-						as.logger.Errorf("AZ : Mismatch in datatype for type : %s, column : %s, value : %s, err : %v", valueType, sortedColumnKeys[index], strValue, err)
-						finalColumnValues = append(finalColumnValues, nil)
-					} else {
-						finalColumnValues = append(finalColumnValues, convertedValue)
-					}
-				case "string":
-					// This is needed to enable diacritic support Ex: Ü,ç Ç,©,∆,ß,á,ù,ñ,ê
-					// A substitute to this PR; https://github.com/denisenkom/go-mssqldb/pull/576/files
-					// An alternate to this approach is to use nvarchar(instead of varchar)
-					if len(strValue) > mssqlStringLengthLimit {
-						strValue = strValue[:mssqlStringLengthLimit]
-					}
-					var byteArr []byte
-					if hasDiacritics(strValue) {
-						as.logger.Debug("diacritics " + strValue)
-						byteArr = str2ucs2(strValue)
-						// This is needed as with above operation every character occupies 2 bytes
-						if len(byteArr) > mssqlStringLengthLimit {
-							byteArr = byteArr[:mssqlStringLengthLimit]
-						}
-						finalColumnValues = append(finalColumnValues, byteArr)
-					} else {
-						as.logger.Debug("non-diacritic : " + strValue)
-						finalColumnValues = append(finalColumnValues, strValue)
-					}
-				default:
-					finalColumnValues = append(finalColumnValues, value)
-				}
-			}
-			// This is needed for the copyIn to proceed successfully for azure synapse else will face below err for missing old columns
-			// mssql: Column count in target table does not match column count specified in input.
-			// If BCP command, ensure format file column count matches destination table. If SSIS data import, check column mappings are consistent with target.
-			for range extraColumns {
-				finalColumnValues = append(finalColumnValues, nil)
-			}
-			_, err = stmt.ExecContext(ctx, finalColumnValues...)
-			if err != nil {
-				as.logger.Errorf("AZ: Error in exec statement for loading in staging table:%s: %v", stagingTableName, err)
-				txn.Rollback()
-				return
-			}
-			csvRowsProcessedCount++
-		}
-		gzipReader.Close()
-		gzipFile.Close()
+	}
+	if _, err = stmt.ExecContext(ctx); err != nil {
+		return nil, "", fmt.Errorf("executing copyIn statement: %w", err)
 	}
 
-	_, err = stmt.ExecContext(ctx)
+	log.Infow("deleting from load table")
```

Comment:
I feel we should have this since it's a critical stage in our loading process. Currently, making these as debug logs. If we feel that some are still polluting it, we can change those later. wdyt?

```
log.Debugw("creating staging table")
log.Debugw("creating prepared stmt for loading data")
log.Debugw("committing transaction")
log.Debugw("waiting for load/append/merge job to complete", "jobID", job.ID())
log.Debugw("job statistics")
log.Debugw("setting search path")
log.Debugw("generated manifest", "manifestLocation", manifestLocation)
```